x-pack/test/apm_api_integration/tests/storage_explorer/storage_details.spec.ts

```diff
@@ -0,0 +1,175 @@
+      endpoint: 'GET /internal/apm/services/{serviceName}/storage_details',
+      params: {
+        path: {
+          serviceName,
+          ...overrides?.path,
+        },
+        query: {
+          indexLifecyclePhase: IndexLifecyclePhaseSelectOption.All,
+          probability: 1,
+          environment: 'ENVIRONMENT_ALL',
+          start: new Date(start).toISOString(),
+          end: new Date(end).toISOString(),
+          kuery: '',
+          ...overrides?.query,
+        },
+      },
+    });
+  }
+
+  registry.when(
+    'Storage details when data is not loaded',
+    { config: 'basic', archives: [] },
+    () => {
+      it('handles empty state', async () => {
+        const { status, body } = await callApi();
+
+        expect(status).to.be(200);
+        expect(body.processorEventStats).to.have.length(4);
+        expect(areProcessorEventStatsEmpty(body)).to.be(true);
+      });
+    }
+  );
+
+  registry.when('Storage details', { config: 'basic', archives: [] }, () => {
+    describe('when data is loaded', () => {
+      before(async () => {
+        const serviceGo = apm
+          .service({ name: serviceName, environment: 'production', agentName: 'go' })
+          .instance('instance');
+
+        await synthtraceEsClient.index([
+          timerange(start, end)
+            .interval('5m')
+            .rate(1)
+            .generator((timestamp) =>
+              serviceGo
+                .transaction({ transactionName: 'GET /api/product/list' })
+                .duration(2000)
+                .timestamp(timestamp)
+                .children(
+                  serviceGo
+                    .span({ spanName: '/_search', spanType: 'db', spanSubtype: 'elasticsearch' })
+                    .destination('elasticsearch')
+                    .duration(100)
+                    .success()
+                    .timestamp(timestamp),
+                  serviceGo
+                    .span({ spanName: '/_search', spanType: 'db', spanSubtype: 'elasticsearch' })
+                    .destination('elasticsearch')
+                    .duration(300)
+                    .success()
+                    .timestamp(timestamp)
+                )
+                .errors(
+                  serviceGo.error({ message: 'error 1', type: 'foo' }).timestamp(timestamp),
+                  serviceGo.error({ message: 'error 2', type: 'foo' }).timestamp(timestamp),
+                  serviceGo.error({ message: 'error 3', type: 'bar' }).timestamp(timestamp)
+                )
+            ),
+        ]);
+      });
+
+      after(() => synthtraceEsClient.clean());
+
+      it('returns correct stats for processor events', async () => {
+        const { status, body } = await callApi();
+        expect(status).to.be(200);
+
+        expect(body.processorEventStats).to.have.length(4);
+
+        const transactionStats = body.processorEventStats.find(
+          ({ processorEvent }) => processorEvent === ProcessorEvent.transaction
+        );
```

Comment:
I think you could use [keyBy](https://lodash.com/docs/4.17.15#keyBy) to create an object per processorEvent. With that, you won't need to use find for each type.

```suggestion
        const statsGroupedByEvent = keyBy(body.processorEventStats, 'processorEvent')
        const transactionStats = statsGroupedByEvent[ProcessorEvent.transaction]
        const spanStats = statsGroupedByEvent[ProcessorEvent.span]
        ....
```

Reply:
Makes sense, added it in https://github.com/elastic/kibana/pull/141391/commits/b8679d235120a186f521b07891eba768e504adf6